File: src/main/java/com/lcaohoanq/nocket/domain/reaction/PostReaction.java
package com.lcaohoanq.nocket.domain.reaction;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.lcaohoanq.nocket.base.entity.BaseEntity;
import com.lcaohoanq.nocket.domain.post.Post;
import com.lcaohoanq.nocket.domain.user.User;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "post_reactions")
public class PostReaction extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "post_id", nullable = false)
    @JsonBackReference
    private Post post;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonManagedReference(value = "user-reactions")
    private User user;

    @ManyToOne
    @JoinColumn(name = "reaction_id", nullable = false)
    @JsonManagedReference(value = "reaction-reactions")
    private Reaction reaction;
    
}

File: src/main/java/com/lcaohoanq/nocket/domain/reaction/PostReaction.kt
package com.lcaohoanq.nocket.domain.reaction

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonManagedReference
import com.lcaohoanq.nocket.base.entity.BaseEntity
import com.lcaohoanq.nocket.domain.post.Post
import com.lcaohoanq.nocket.domain.user.User
import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "post_reactions")
class PostReaction : BaseEntity() {
    @ManyToOne
    @JoinColumn(name = "post_id", nullable = false)
    @JsonBackReference
    var post: Post? = null

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonManagedReference(value = "user-reactions")
    var user: User? = null

    @ManyToOne
    @JoinColumn(name = "reaction_id", nullable = false)
    @JsonManagedReference(value = "reaction-reactions")
    var reaction: Reaction? = null
}
